tdp/core/variables/variables.py

```diff
@@ -118,22 +118,25 @@ def __enter__(self):
         return proxy(self)
 
     def __exit__(self, exc_type, exc_val, exc_tb):
-        self.close()
+        self._close()
 
     def _flush_on_disk(self) -> None:
         """Write the content of the variables file on disk.
 
         Raises:
             RuntimeError: If the file descriptor is already closed.
         """
+        # Check if the file descriptor is already closed
         if not self._file_descriptor or self._file_descriptor.closed:
             raise RuntimeError(
                 f"{self._file_path} is already closed, which shouldn't happen"
             )
 
+        # Check if the file is writable
         if not self._file_descriptor.writable():
```

Comment:
If `_flush_on_disk` is only used in `_close`, this condition is never passing, because this is already checked [here](https://github.com/TOSIT-IO/tdp-lib/pull/522/files#diff-60263f9c86f45f8ceaaabbdb576ec29dd446d11c610fa7d7e114bafd5f925280R162)

Reply:
Agree, I left it to prevent a bad usage of  `_flush_on_disk`.